core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java

```diff
@@ -358,13 +358,21 @@ default void checkCanShowColumns(SystemSecurityContext context, CatalogSchemaTab
     }
 
     /**
-     * Filter the list of columns to those visible to the identity.
+     * Filter the list of columns to those visible to the identity when querying metadata.
      */
     default Set<String> filterColumns(SystemSecurityContext context, CatalogSchemaTableName table, Set<String> columns)
     {
         return emptySet();
     }
 
+    /**
+     * Filter the list of columns to those visible to the identity when querying the table.
+     */
+    default Set<String> filterTableSchema(SystemSecurityContext context, CatalogSchemaTableName table, Set<String> columns)
+    {
+        return emptySet();
```

Comment:
it should return all columns, so by default it will behave following ANSI SQL behavior. Notice that default implementation `checkCanSelect` will make sure that by default access is restricted.